solver/src/driver.rs

```diff
@@ -309,6 +312,9 @@ impl Driver {
         futures::stream::iter(settlements)
             .filter_map(|settlement| async {
                 let surplus = settlement.settlement.total_surplus(prices);
+                // Given the fee discount the solver fees by themselves may not be sufficient to make a solution economically viable (leading to a negative objective value)
+                // We therefore reverse apply the fee discount to simulate unsubsidized fees for ranking.
+                let solver_fees = settlement.settlement.total_fees(prices) / BigRational::from_float(self.fee_discount_factor).expect("Discount factor is not a rational");
```

Comment:
If you'd like to make it more explicit, I was having this as a SettlementEncoder member called `unsubsidized_fees` in the other [PR](https://github.com/gnosis/gp-v2-services/pull/598/files#diff-2c61b8ac9597c7fc4f938bf785c1f2b9777cd3b65d04f444be9860e30364f0a1R223-R230).